BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -323,9 +326,58 @@ public void UpdatePost(Post newPost, Post oldPost)
             SavePost(newPost);
         }
 
-        public bool CheckSlugExists(string slug)
+        public void SaveFiles(List<IFormFile> files)
+        {           
+            foreach(var file in files)
+            {
+                if(file.Length > 0)
+                {                    
+                    using (Stream uploadedFileStream = file.OpenReadStream())
+                    {
+                        byte[] buffer = new byte[uploadedFileStream.Length];
+                        uploadedFileStream.Read(buffer, 0, buffer.Length);
+                        string name = CreateFileName(file.FileName);
+                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<string> GetFileNames()
+        {
+            IEnumerable<string> fileNames = _fileSystem.EnumerateFiles(UploadsFolder);
+            return fileNames;
+        }
+
+        private bool CheckFileNameExists(string fileName)
         {
-            return _fileSystem.FileExists($"{PostsFolder}\\{slug}.xml");
+            return _fileSystem.FileExists($"{UploadsFolder}\\{fileName}");
         }
+
+        private string CreateFileName(string fileName)
+        {
+            string tempName = fileName;
+            string[] elements = fileName.Split(".");
+            int count = 0;
+            while (CheckFileNameExists(tempName))
+            {
+                count++;

```

Comment:
Wouldn't it be easier with count = 0 so you increment at the end of the loop as opposed to the beginning. That way the first item is -0 and then -1 .... ?

Reply:
@lucasisaza To me it makes more sense to append `-1` to a duplicate file name rather than `-0`. Also I incremented `count` at the beginning to make sure that when the loop breaks, `count` stays at the number that it was set to, in case we need to use it for anything else.

Reply:
I agree that starting at `-1` is better, but it feels a little weird how you're using count.  Also, I wouldn't worry now about whether we need to use it later on (YAGNI).

Aesthetics vary, but another way you could do this is with a for loop:

    for(int i = 1; CheckFileNameExists(tempName); i++)
    { /* ... */ }